lib/vintage_net_wizard/web/endpoint.ex

```diff
@@ -58,33 +61,27 @@ defmodule VintageNetWizard.Web.Endpoint do
     ]
   end
 
-  defp maybe_use_ssl(_use_ssl = true) do
-    ssl_dir = ssl_dir()
+  defp maybe_use_ssl(_use_ssl = true, opts) do
+    port = Application.get_env(:vintage_net_wizard, :port, 443)
+    ssl_options = Keyword.get(opts, :ssl)
+    options = [dispatch: dispatch(), port: port]
 
     Plug.Cowboy.child_spec(
       plug: Router,
       scheme: :https,
-      options: [
-        dispatch: dispatch(),
-        certfile: Application.get_env(:vintage_net_wizard, :certfile, "#{ssl_dir}/cert.pem"),
-        keyfile: Application.get_env(:vintage_net_wizard, :keyfile, "#{ssl_dir}/key.pem"),
-        port: Application.get_env(:vintage_net_wizard, :port, 443)
-      ]
-    )
-  end
-
-  defp maybe_use_ssl(_no_ssl) do
-    Plug.Cowboy.child_spec(
-      plug: Router,
-      scheme: :http,
-      options: [
-        dispatch: dispatch(),
-        port: Application.get_env(:vintage_net_wizard, :port, 80)
-      ]
+      options: Keyword.merge(ssl_options, options)
```

Comment:
We might need to swap this since `:port` is a valid ssl option and providing that will always be overwritten by the defaults here?
```suggestion
      options: Keyword.merge(options, ssl_options)
```

Reply:
I did this to avoid anyone passing in an option in the `:ssl` opts that would override our defined options.